compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -479,6 +479,16 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         let attrs = std::mem::take(&mut self.attrs);
         let mut bodies = std::mem::take(&mut self.bodies);
         let local_node_ids = std::mem::take(&mut self.local_node_ids);
+
+        let local_id_to_def_id = local_node_ids
+            .iter()
+            .filter_map(|&node_id| {
+                let def_id = self.resolver.opt_local_def_id(node_id)?;
```

Comment:
Try adding the filter for ID zero in this PR, too

Generally the PR looks good, so the regression must be in the third commit of the original PR